### DISCUSSION TITLE
Support additional way to display summary of blog post.

### DISCUSSION THREAD

**Discussion by Ouch1978** [facebook/docusaurus#5043] [Ideas]
The original summary truncate function with  tag is good.
But I think it would be better if there are additional ways to set the summary, for example:
1. Take the description attribute in header as summary.
2. Provide an additional property to set default lines as summary, for example set the value to 10 to show first 10 lines as summary.

**Comment by slorber** [comment]
For retrocompatibility, it is not easy to use the description as summary (breaking change).

For now the simplest would probably be to allow a `summary` frontmatter. It may not be the most convenient for all use-cases but it is flexible

**Reply by sparanoid** [reply]
That would be great if we can add a `summary` or `description` frontmatter to show on every post. This is common from a Jekyll migrated site and I also see docs using summary seems useful:

![image](https://user-images.githubusercontent.com/96356/126662808-56271807-61c1-48d2-a446-71163c502462.png)
One sentence summary screenshot from Stripe documentation.

However there's no simple way to do that unless we swizzle the commponent (`DocItem` or `DocPage`? which are marked as unsafe).
